golem-worker-service/src/grpcapi/worker.rs

```diff
@@ -331,6 +353,42 @@ impl WorkerGrpcApi {
         Ok(metadata.into())
     }
 
+    async fn get_worker_metadatas(
+        &self,
+        request: GetWorkerMetadatasRequest,
+    ) -> Result<(Option<u64>, Vec<WorkerMetadata>), GrpcWorkerError> {
+        let template_id: golem_common::model::TemplateId = request
+            .template_id
+            .ok_or_else(|| bad_request_error("Missing template id"))?
+            .try_into()
+            .map_err(|_| bad_request_error("Invalid template id"))?;
+
+        let filter: Option<WorkerFilter> =
+            match request.filter {
+                Some(f) => Some(f.try_into().map_err(|error| {
```

Comment:
Same here. I strongly feel its a overuse of `TryInto`, `TryFrom`

Reply:
not sure what do you mean by that